Purchases/Public/Purchases+async.swift

```diff
@@ -0,0 +1,271 @@
+                }
+                guard let customerInfo = maybeCustomerInfo else {
+                    fatalError("Expected non-nil result 'customerInfo' for nil error")
+                }
+                continuation.resume(returning: customerInfo)
+            }
+        }
+    }
+
+    @available(iOS 15.0, macOS 12, tvOS 15.0, watchOS 8.0, *)
+    func restoreTransactionsAsync() async throws -> CustomerInfo {
+        return try await withCheckedThrowingContinuation { continuation in
+            restoreTransactions { maybeCustomerInfo, maybeError in
+                if let error = maybeError {
+                    continuation.resume(throwing: error)
+                    return
+                }
+                guard let customerInfo = maybeCustomerInfo else {
+                    fatalError("Expected non-nil result 'customerInfo' for nil error")
+                }
+                continuation.resume(returning: customerInfo)
+            }
+        }
+    }
+
+    @available(iOS 15.0, macOS 12, tvOS 15.0, watchOS 8.0, *)
+    func checkTrialOrIntroductoryPriceEligibilityAsync(_ productIdentifiers: [String]) async
+    -> [String: IntroEligibility] {
+        return await withCheckedContinuation { continuation in
+            checkTrialOrIntroductoryPriceEligibility(productIdentifiers) { result in
+                continuation.resume(returning: result)
+            }
+        }
+    }
+
+    @available(iOS 15.0, macOS 12, tvOS 15.0, watchOS 8.0, *)
+    func paymentDiscountAsync(forProductDiscount discount: SKProductDiscount,
+                              product: SKProduct) async throws -> SKPaymentDiscount {
+        return try await withCheckedThrowingContinuation { continuation in
+            paymentDiscount(forProductDiscount: discount, product: product) { maybeDiscount, maybeError in
+                if let error = maybeError {
+                    continuation.resume(throwing: error)
+                    return
+                }
+                guard let discount = maybeDiscount else {
+                    fatalError("Expected non-nil 'discount' for nil error")
+                }
+                continuation.resume(returning: discount)
+            }
+        }
+    }
+
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    @available(iOS 15.0, macOS 12, *)
+    func showManageSubscriptionModalAsync() async throws {
+        return try await withCheckedThrowingContinuation { continuation in
+            showManageSubscriptionModal { error in
+                if let error = error {
+                    continuation.resume(throwing: error)
+                    return
+                }
+                continuation.resume(returning: ())
+            }
+        }
+    }
+
+    @available(iOS 15.0, macOS 12.0, *)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    @objc func beginRefundRequestAsync(for productID: String) async throws -> RefundRequestStatus {
```

Comment:
I'm having a bit of a hard time figuring out API availability here: 
With this setting ^ this won't compile on tvOS, stating that Swift concurrency is available only on tvOS 15.0. But I'm trying to make it _unavailable_ entirely. Not sure why it's even trying to use this for tvOS. 
I get the same issue if I mark this unavailable for macOS (which it should be, but it wasn't marked before this PR, I'd like to fix that before wrapping up this PR). 

Any thoughts here?

Reply:
> With this setting ^ this won't compile on tvOS, stating that Swift concurrency is available only on tvOS 15.0. But I'm trying to make it unavailable entirely. Not sure why it's even trying to use this for tvOS.

Yeah `@available` is so limited. I think you need to specify the version and make it unavailable:
```swift
@available(iOS 15.0, macOS 12, tvOS 15.0, watchOS 8.0, *)
@available(watchOS, unavailable)
@available(tvOS, unavailable)
```
Does that work?

Reply:
I tried it and I get 
```ruby
 - WARN  | [OSX] xcodebuild:  .../RevenueCat-Swift.h:1158:243: warning: availability does not match previous declaration
 - NOTE  | [OSX] xcodebuild: .../RevenueCat-Swift.h:1158:365: note: previous attribute is here